builtin/providers/azurerm/resource_arm_network_interface_card.go

```diff
@@ -252,6 +258,17 @@ func resourceArmNetworkInterfaceRead(d *schema.ResourceData, meta interface{}) e
 		}
 	}
 
+	if iface.IPConfigurations != nil && len(*iface.IPConfigurations) > 0 {
+		var privateIPAddress *string
+		for _, ipConfig := range *iface.IPConfigurations {
+			privateIPAddress = ipConfig.Properties.PrivateIPAddress
```

Comment:
Hi @Felivel 

The issue here is that you cannot index when a pointer is in play. So it has to be dereferrenced like this:

```
(*iface.IPConfigurations)[0].Properties.PrivateIPAddress
```